### PR TITLE
Add access_control rule for form login auth

### PR DESCRIPTION
**Pull request by GaryPEGEOT**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #396 
| License       | MIT

Add `access_control` rule when `make:auth` command is used with login form authenticator

### CODE REVIEW

**Comment by romaricdrigon** [comment]
Hello,
Thank you for this PR, sorry for the delay in reviewing it.
The feature sounds nice. Does it support merging rules with already existing acces_control rules? We should not break user config.

**Comment by GaryPEGEOT** [comment]
> 
> 
> Hello,
> Thank you for this PR, sorry for the delay in reviewing it.
> Does it support merging rules with already existing acces_control rules? We should not break user config.

Hi! Yes, it's [done here](https://github.com/symfony/maker-bundle/pull/460/files#diff-f6787c726ea39bb082882861822a762bR98-R103): If an existing config is detected, the line is not added. The test case name is `simple_security_with_access_control`

File: src/Security/SecurityConfigUpdater.php [hunk 1]
```diff
@@ -92,6 +92,21 @@ public function updateForAuthenticator(string $yamlSource, string $firewallName,
         }
 
         $newData['security']['firewalls'][$firewallName] = $firewall;
+
+        $accessControlRules = $newData['security']['access_control'] ?? [];
+
+        foreach ($accessControlRules as $rule) {
+            if ('^/login$' === $rule['path']) {
```
Comment (romaricdrigon) [inline src/Security/SecurityConfigUpdater.php]: I think that check is a little bit too strict: I think we should check for `^/login`, it's the syntax from the (old) documentation: https://symfony.com/doc/4.0/security/form_login_setup.html#be-sure-the-login-page-isn-t-secure-redirect-loop

File: tests/Security/yaml_fixtures/expected_authenticator/simple_security_with_added_access_control.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        dev: ~
+        main:
+            anonymous: true
+            guard:
+                authenticators:
+                    - App\Security\AppCustomAuthenticator
+    access_control:
+        -
+            path: ^/login$
+            roles: IS_AUTHENTICATED_ANONYMOUSLY
```
Comment (romaricdrigon) [inline tests/Security/yaml_fixtures/expected_authenticator/simple_security_with_added_access_control.yaml]: Is there any way to use the inline syntax (as in `simple_security_with_access_control.yaml`) instead of that one?

**Review by romaricdrigon — CHANGES_REQUESTED** [review]
Great, we are almost there!

File: src/Security/SecurityConfigUpdater.php (hunk 1 above)
Comment (GaryPEGEOT) [reply src/Security/SecurityConfigUpdater.php]: Fixed!

File: tests/Security/yaml_fixtures/expected_authenticator/simple_security_with_added_access_control.yaml (hunk 2 above)
Comment (GaryPEGEOT) [reply tests/Security/yaml_fixtures/expected_authenticator/simple_security_with_added_access_control.yaml]: Not that I'm aware of. It was pointed out in the [related issue](https://github.com/symfony/maker-bundle/issues/396#issuecomment-496578320) that this might happen

Comment (romaricdrigon) [reply tests/Security/yaml_fixtures/expected_authenticator/simple_security_with_added_access_control.yaml]: Ok, thank you for the clarification!

**Review by romaricdrigon — APPROVED** [review]

**Comment by weaverryan** [comment]
Hiya!

This generally look very good (including the tests). But, I think I need to do a bit of work on `YamlSourceManipulator` to allow for the `access_control` to be inline as we really want. Also, I find a few edge-cases with it - like if the `security.yaml` file looks like the recipe - https://github.com/symfony/recipes/blob/master/symfony/security-bundle/4.4/config/packages/security.yaml - where it has `access_control` set to no value, then with 2 lines of comments, those comments end up in a screwy place. I'd like to get that done, then this PR (which is already perfect for what it does) should be ready

**Review by maxhelias — APPROVED** [review]

**Comment by maxhelias** [comment]
A rebase is required

**Comment by GaryPEGEOT** [comment]
> A rebase is required

Rebase done !
